test/missing.py

```diff
@@ -367,6 +367,30 @@ def test_avail_subclasses_in(self):
         # cleanup
         sys.path = orig_sys_path
 
+    def test_shell_quote(self):
+        """Test shell_quote function"""
+        values = [
+            (123, "'123'", True),
+            ('foo', "'foo'", True),
+            ('value with whitespace', "'value with whitespace'", True),
+            ('foo\tbar', "'foo\tbar'", True),
+            ('(value)', "'(value)'", True),
+            ('$value', "'$value'", True),
+            ('value with (foo)', "'value with (foo)'", True),
+            ('value with $foo', "'value with $foo'", True),
+            # check that escaped single quotes aren't escaped again
+            # shell_unqoute can't unquote a single-quoted string with (single) quotes in it (ValueError: No closing quotation)
+            ("foo'bar\\'baz", "'foo\\'bar\\'baz'", False),
+            ("'", "'\\''", False),
+            ("\'", "'\\''", False),
+            ("\\'", "'\\''", False),
+            ("''", "'\\'\\''", False),
+        ]
+        for orig_value, quoted_value, test_unquote in values:
+            self.assertEqual(shell_quote(orig_value), quoted_value)
+            if test_unquote:
+                self.assertEqual(str(orig_value), shell_unquote(shell_quote(orig_value)))
+
```

Comment:
if you're sure you want the value error to be ok, then you have to test for it, so add an else case and assert that it raises the error.